internal/mode/static/nginx/runtime/manager.go

```diff
@@ -151,18 +188,27 @@ func (m *ManagerImpl) GetUpstreams() (ngxclient.Upstreams, error) {
 	return *upstreams, nil
 }
 
-// EnsureNginxRunning ensures NGINX is running by locating the main process.
-func EnsureNginxRunning(ctx context.Context) error {
-	if _, err := findMainProcess(ctx, os.Stat, os.ReadFile, pidFileTimeout); err != nil {
+type ProcessHandlerImpl struct {
+	readFile  ReadFileFunc
+	checkFile CheckFileFunc
+}
+
+func NewProcessHandlerImpl(readFileFunc ReadFileFunc, checkFileFunc CheckFileFunc) *ProcessHandlerImpl {
+	return &ProcessHandlerImpl{readFile: readFileFunc, checkFile: checkFileFunc}
+}
+
+// FindProcess ensures NGINX is running by locating the main process.
+func (p *ProcessHandlerImpl) FindProcess(ctx context.Context) error {
```

Comment:
You can remove this function and just keep the `FindMainProcess` function below.

Reply:
done, removed that function